src/Foundatio/Storage/FolderFileStorage.cs

```diff
@@ -32,31 +32,47 @@ public class FolderFileStorage : IFileStorage {
                 folder += Path.DirectorySeparatorChar;
 
             Folder = folder;
-            
+
             _logger.LogInformation("Creating {Directory} directory", folder);
             Directory.CreateDirectory(folder);
         }
 
-        public FolderFileStorage(Builder<FolderFileStorageOptionsBuilder, FolderFileStorageOptions> config) 
+        public FolderFileStorage(Builder<FolderFileStorageOptionsBuilder, FolderFileStorageOptions> config)
             : this(config(new FolderFileStorageOptionsBuilder()).Build()) { }
 
         public string Folder { get; set; }
         ISerializer IHaveSerializer.Serializer => _serializer;
 
-        public Task<Stream> GetFileStreamAsync(string path, CancellationToken cancellationToken = default) {
+        public Task<Stream> GetFileStreamAsync(string path, CancellationToken cancellationToken = default)
+            => GetFileStreamAsync(path, FileAccess.Read, cancellationToken);
+
+        public Task<Stream> GetFileStreamAsync(string path, FileAccess fileAccess, CancellationToken cancellationToken = default) {
```

Comment:
I like this signature. I would say that we should mark the current method as deprecated so that we can get rid of it on the next major version.